Solution/source/Menu/Routine.cpp

```diff
@@ -129,25 +140,11 @@ void ThreadMenyooMain()
 	MenyooMain();
 }
 
-void ThreadMenyooConfig()
+void TickMenyooConfig()
```

Comment:
The if statement on line 145 and 147 should switch place, there is no need for the if statement on line 160 to be nesteled 😄 

**Like this:**
```
void TickMenyooConfig()
{
	if ( g_ConfigHasNotBeenRead )
	{
		if (GetTickCount() > g_MenyooConfigOnceTick + 9000)
		{
			if (MenuConfig::ConfigInit() < 0)
			{
				ige::myLog << ige::LogType::LOG_ERROR << "Failed to load menyooConfig from " << GetPathff(Pathff::Main, true) << "menyooConfig.ini.";
			}
			else
			{
				MenuConfig::ConfigRead();
			}
			g_ConfigHasNotBeenRead = false;
		}
	}

	if (GetTickCount() > g_MenyooConfigTick + 30000)
	{
		if (MenuConfig::bSaveAtIntervals)
		{
			MenuConfig::ConfigSave();
		}
		g_MenyooConfigTick = GetTickCount();
	}
}
```

Reply:
This was done to keep the "block everything for 9 seconds, and from then on do the check once, and then save the config every 30 seconds". I'm fine with either solution, just let me know if the change in behavior is fine.